packages/block-editor/src/components/rich-text/use-input-rules.js

```diff
@@ -13,6 +18,34 @@ import { useDispatch } from '@wordpress/data';
 import { store as blockEditorStore } from '../../store';
 import { preventEventDiscovery } from './prevent-event-discovery';
 
+// A robust way to retain selection position through various
+// transforms is to insert a special character at the position and
+// then recover it.
+const START_OF_SELECTED_AREA = '\u0086';
+
+function findSelection( blocks ) {
```

Comment:
While this is the only use case for nested blocks, there may be value in reusing this function in other places where we use the `START_OF_SELECTED_AREA` trick.

Reply:
I took advantage of this and abstracted it into a reusable function while removing Lodash in https://github.com/WordPress/gutenberg/pull/41806. Would love to get your feedback there when you get a chance @ellatrix.